sentry_sdk/scope.py

```diff
@@ -1259,6 +1354,61 @@ def _apply_contexts_to_event(self, event, hint, options):
                 "replay_id": replay_id,
             }
 
+    def _drop(self, cause, ty):
+        # type: (Any, str) -> Optional[Any]
+        logger.info("%s (%s) dropped event", ty, cause)
+        return None
+
+    def run_error_processors(self, event, hint):
+        # type: (Event, Hint) -> Optional[Event]
+        """
+        Runs the error processors on the event and returns the modified event.
+        """
+        exc_info = hint.get("exc_info")
+        if exc_info is not None:
+            error_processors = chain(
+                Scope.get_global_scope()._error_processors,
+                Scope.get_isolation_scope()._error_processors,
+                Scope.get_current_scope()._error_processors,
+            )
+
+            for error_processor in error_processors:
+                new_event = error_processor(event, exc_info)
+                if new_event is None:
+                    return self._drop(error_processor, "error processor")
+
+                event = new_event
+
+        return event
+
+    def run_event_processors(self, event, hint):
+        # type: (Event, Hint) -> Optional[Event]
+        """
+        Runs the event processors on the event and returns the modified event.
+        """
+        ty = event.get("type")
+        is_check_in = ty == "check_in"
+
+        if not is_check_in:
+            isolation_scope = _isolation_scope.get()
+            current_scope = _current_scope.get()
```

Comment:
Is it intentional that we're using the context vars directly here but in `run_error_processors` we're using the `Scope.get_{type}_scope` proxies? Can we unify this?

Reply:
Yes this is intentional. If I use the proxies then if no scope exists it is created and we end up in an infinite loop (I think it was because the auto enabled logging captures logs that are printed during creation of scopes and then it creates again scopes and we are back here. or something like this.)